msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
> Well, if you feel strongly about it, happy to drop that functionality, although I think your solution to this use case should be documented. But I don't quite see how your solution works? What happens to all the intermediate internal nodes between 0 and 1?

The nodes iterator returns all nodes in the subtree u in preorder. So, all branches in the subtree will be coloured. Try it out.

> More generally, if you can suggest an easy msprime way to calculate the nodes I want to colour, then fine. You have a better grasp of the msprime functions than me, so perhaps you could suggest? I guess you know what I want to do: take (say) nodes {1,2,5,6,9,10} and colour in such a way that we can see immediately whether they cluster. This requires working out which nodes give rise to only those tips, and colouring those with the same colour.

You can colour them using precisely the same algorithm in client code. Please just remove this complexity, I'm not going to merge the patch otherwise.

We can make a cookbook of clever ways to colour trees in the documentation, and there will be myriad of different ways that people will want to do it.